hedera-node/hedera-mono-service/src/main/java/com/hedera/services/txns/crypto/AutoCreationLogic.java

```diff
@@ -240,7 +259,11 @@ public Pair<ResponseCodeEnum, Long> create(
         // If the transaction fails, we will get an opportunity to unlink this alias in
         // reclaimPendingAliases()
         aliasManager.link(alias, EntityNum.fromAccountId(newId));
-        aliasManager.maybeLinkEvmAddress(jKey, EntityNum.fromAccountId(newId));
+        if (alias.size() > EntityIdUtils.EVM_ADDRESS_SIZE) {
```

Comment:
Probably you can move this `if` body logic in the `else` structure above, on line `237`? In this way you wouldn't have a double check for the same clause and duplicated `JKey` initialization.

Reply:
Here we need the `newId` variable that is created after the `else` clause above so it's not straightforward to move this `if` body.